lib/types.bzl

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Skylib module containing functions checking types."""
+
+
+# create instance singletons to avoid unnecessary allocations
+_a_bool = True
+_a_dict = {}
+_a_list = []
+_a_string = ""
+_a_tuple = ()
+_an_int = 1
+
+
+def _a_function():
+  pass
+
+
+def _is_list(v):
+    """Returns True if v is an instance of the list.
```

Comment:
Super-nit: Here and in the `Returns` clause, can you say "a list", etc. instead of "the list"? (Throughout the docs for each of these functions.)